internal/provider/resources/code_location.go

```diff
@@ -125,18 +216,39 @@ func (r *CodeLocationResource) Read(ctx context.Context, req resource.ReadReques
 			tflog.Trace(ctx, "Code Location not found, probably already deleted manually, removing from state")
 			resp.State.RemoveResource(ctx)
 		} else {
-			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read team, got error: %s", err))
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read code locations, got error: %s", err))
 		}
 		return
 	}
 
-	attributeTypes := map[string]attr.Type{
-		"python_file": types.StringType,
+	// Code source
+	codeSourceAttributeTypes := map[string]attr.Type{
+		"module_name":  types.StringType,
+		"package_name": types.StringType,
+		"python_file":  types.StringType,
+	}
+	codeSourceAttributeValues := map[string]attr.Value{
+		"module_name":  types.StringValue(codeLocation.CodeSource.ModuleName),
+		"package_name": types.StringValue(codeLocation.CodeSource.PackageName),
+		"python_file":  types.StringValue(codeLocation.CodeSource.PythonFile),
+	}
+	codeSource, diag := types.ObjectValue(codeSourceAttributeTypes, codeSourceAttributeValues)
+	resp.Diagnostics.Append(diag...)
+
+	if resp.Diagnostics.HasError() {
```

Comment:
Is returning correct? That leaves the resource half-"read"?

Reply:
https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types#ObjectValueMust

> ObjectValueMust creates a Object with a known value, converting any diagnostics into a panic at runtime

Reply:
rather have a graceful error than a panic tbh